Sources/NIOTCPEchoServer/Server.swift

```diff
@@ -0,0 +1,125 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2023 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+#if swift(>=5.9)
+@_spi(AsyncChannel) import NIOCore
+@_spi(AsyncChannel) import NIOPosix
+
+@available(macOS 14, *)
+@main
+struct Server {
+    /// The server's host.
+    private let host: String
+    /// The server's port.
+    private let port: Int
+    /// The server's event loop group.
+    private let eventLoopGroup: MultiThreadedEventLoopGroup
+
+    static func main() async throws {
+        let server = Server(
+            host: "localhost",
+            port: 8765,
+            eventLoopGroup: .singleton
+        )
+        try await server.run()
+    }
+
+    /// This method starts the server and handles incoming connections.
+    func run() async throws {
+        let channel = try await ServerBootstrap(group: eventLoopGroup)
+            .serverChannelOption(ChannelOptions.socketOption(.so_reuseaddr), value: 1)
+            .bind(
+                host: self.host,
+                port: self.port
+            ) { channel in
+                channel.eventLoop.makeCompletedFuture {
+                    // We are using two simple handlers here to frame our messages with "\n"
+                    try channel.pipeline.syncOperations.addHandler(ByteToMessageHandler(NewlineDelimiterCoder()))
```

Comment:
all these `try`s should be `try!`s, they can't fail unless programmer error, right?

Reply:
They could throw if the channel is closed while we add the handler. I don't wanna show adopters a `try!` when it could fail in rare cases and I don't think the `try` is too bad.